waku/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -855,7 +855,8 @@ method isReady*(g: OnchainGroupManager): Future[bool] {.async.} =
   g.retryWrapper(currentBlock, "Failed to get the current block number"):
     cast[BlockNumber](await g.ethRpc.get().provider.eth_blockNumber())
 
-  if g.latestProcessedBlock < currentBlock:
+  # the node is still able to process messages if it is behind the latest block by a factor of the valid roots
+  if uint(g.latestProcessedBlock) < uint(currentBlock) - uint(g.validRoots.len):
```

Comment:
isn't it better widening rather than narrowing? i mean instead of casting everything to unit and operating with that, using the `latestProcessedBlock` type which has more bits and then "widening" validRoots to do the `-`?

wondering also if in the first run we can hit `currentBlock-g.validRoots.len` where `currentBlock=0` and since we are using unit that will lead to a huge value, meaning that we are ready when we are not?

Reply:
currentBlock will never be 0, since that is fetched from the chain (and we can operate under the assumption that it will always be greater than 0). good point about widening vs narrowing

Reply:
casted to `u256` - 442ca43a